### PR TITLE
Idol backend/update subteam

### PR DESCRIPTION
**Pull request by gracshan**
### Summary 

This pull request updates the subteam field for members when a team is edited on IDOL. It also allows for creating new teams and deleting existing teams.

### Test Plan 
Initially, subteam is idol: 
Screen Shot 2021-04-16 at 10 57 17 PM
Adding new team:
Screen Shot 2021-04-16 at 10 54 52 PM
Updates in Firestore:
Screen Shot 2021-04-16 at 10 56 28 PM
Deleting the team replaces subteam field with empty string:
Screen Shot 2021-04-17 at 9 50 27 AM

### CODE REVIEW

File: backend/src/dao/TeamsDao.ts
Comment (SamChou19815) [inline backend/src/dao/TeamsDao.ts:1]: Remove this

File: backend/src/teamAPI.ts
Comment (SamChou19815) [inline backend/src/teamAPI.ts:34]: You can do `const updatedMember = { ...member, subteam: team.name };`

File: dti-website/src/data/all-members.json
Comment (SamChou19815) [inline dti-website/src/data/all-members.json:9]: Revert changes in this file unless the data is good.

File: backend/src/teamAPI.ts
Comment (SamChou19815) [inline backend/src/teamAPI.ts:36]: It should be done using promise, like: 

```typescript
await Promise.all(newMembers.map(async member => {
  const updatedMember = { ...member, subteam: team.name };
  await MembersDao.setMember(updatedMember.email, updatedMember);
});
```

**Review by SamChou19815 — CHANGES_REQUESTED** [review]

File: dti-website/src/data/all-members.json
Comment (gracshan) [reply dti-website/src/data/all-members.json:9]: Can I ask how I can check whether the data is good? I didn't make these changes, so I'm not quite sure where to look.

Comment (SamChou19815) [reply dti-website/src/data/all-members.json:9]: @gracshan The current data about subteam should be mostly right, so diffs that change subteam field to empty string is definitely wrong. We cannot merge this PR because once we merge it, the team website with the wrong data will be auto-deployed.

Comment (gracshan) [reply dti-website/src/data/all-members.json:9]: @SamChou19815 I see. I'll revert it based on what I see displayed in the changes here. Thank you for letting me know!

File: backend/src/dao/TeamsDao.ts
Comment (SamChou19815) [inline backend/src/dao/TeamsDao.ts:51]: why this change?

Comment (henry-li-06) [reply backend/src/dao/TeamsDao.ts:51]: I think originally in the `updateTeamMembers` function, it checked if `team.uuid` was empty and generated a new one if it was. That way when `setTeam` was called, `team.uuid` would always be set already. I think that caused some linter error though since doing so updated the parameter to the function so the function now makes a copy of the team. So this line should actually probably. be changed back.

**Review by SamChou19815 — APPROVED** [review]

**Pull request merged**
